lib/http-bridge/src/Cardano/Wallet/HttpBridge/Compatibility.hs

```diff
@@ -111,31 +111,36 @@ instance PersistTx (HttpBridge network) where
 -- | Encode a public key to a (Byron / Legacy) Cardano 'Address'. This is mostly
 -- dubious CBOR serializations with no data attributes.
 instance KeyToAddress (HttpBridge 'Testnet) SeqKey where
-    keyToAddress = keyToAddressWith
+    keyToAddress = keyToAddressWith @SeqKey
         (attributesWithProtocolMagic (protocolMagic @'Testnet))
 
 instance KeyToAddress (HttpBridge 'Mainnet) SeqKey where
-    keyToAddress = keyToAddressWith emptyAttributes
+    keyToAddress = keyToAddressWith @SeqKey emptyAttributes
 
 instance KeyToAddress (HttpBridge 'Testnet) RndKey where
-    keyToAddress = rndKeyToAddressWith
-        (attributesWithProtocolMagic (protocolMagic @'Testnet))
+    keyToAddress key = keyToAddressWith @RndKey
+        (rndAttributesWithProtocolMagic (protocolMagic @'Testnet) key)
+        key
 
 instance KeyToAddress (HttpBridge 'Mainnet) RndKey where
-    keyToAddress = rndKeyToAddressWith emptyAttributes
-
-rndKeyToAddressWith :: CBOR.Encoding -> RndKey 'AddressK XPub -> Address
-rndKeyToAddressWith _attrs key =
-    error "rndKeyToAddressWith: unimplemented"
-  where
-    _derPath = encodeDerivationPath key
+    keyToAddress = keyToAddressWith @RndKey emptyAttributes
```

Comment:
Note: These aren't empty on Mainnet for the random keys, they still contain the derivation path.

Reply:
ok, so for Mainnet they will not contain protocol magic but : 
```
mempty
    <> CBOR.encodeMapLen 1
    <> CBOR.encodeWord 1
    <> CBOR.encodeBytes (CBOR.toStrictByteString $ encodeDerivationPath key)
```
???

Reply:
corrected